# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../uclibc/patches/ash-uclibc.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---
--- ash/jobs.c.orig	2005-01-21 21:32:31.000000000 -0800
+++ ash/jobs.c	2005-01-21 21:38:23.000000000 -0800
@@ -274,8 +274,8 @@
 				if ((i & 0xFF) == 0177)
 					i >>= 8;
 #endif
-				if ((i & 0x7F) < NSIG && sys_siglist[i & 0x7F])
-					scopy(sys_siglist[i & 0x7F], s);
+				if ((i & 0x7F) < NSIG && strsignal(i & 0x7F))
+					scopy(strsignal(i & 0x7F), s);
 				else
 					fmtstr(s, 64, "Signal %d", i & 0x7F);
 				if (i & 0x80)
@@ -744,8 +744,8 @@
 			if (status == SIGTSTP && rootshell && iflag)
 				outfmt(out2, "%%%d ", job - jobtab + 1);
 #endif
-			if (status < NSIG && sys_siglist[status])
-				out2str(sys_siglist[status]);
+			if (status < NSIG && strsignal(status))
+				out2str(strsignal(status));
 			else
 				outfmt(out2, "Signal %d", status);
 			if (core)
